lib/pat.c

```diff
@@ -1788,31 +1799,32 @@ _grn_pat_create(grn_ctx *ctx,
   grn_io *io;
   pat_node_common *node0;
   struct grn_pat_header *header;
-  uint32_t entry_size, w_of_element;
+  uint32_t entry_size;
   grn_encoding encoding = ctx->encoding;
-  if (flags & GRN_OBJ_KEY_WITH_SIS) {
-    entry_size = sizeof(sis_node) + value_size;
-  } else {
-    entry_size = value_size;
-  }
-  for (w_of_element = 0; (uint32_t)(1 << w_of_element) < entry_size;
-       w_of_element++) {
-    /* nop */
-  }
   {
     grn_io_array_spec array_spec[3];
+    uint32_t segment_width_of_pat_entry;
     array_spec[SEGMENT_KEY].w_of_element = 0;
     if (flags & GRN_OBJ_KEY_LARGE) {
       array_spec[SEGMENT_KEY].max_n_segments = GRN_PAT_MAX_N_SEGMENTS_LARGE;
-      array_spec[SEGMENT_PAT].w_of_element = 5;
-      array_spec[SEGMENT_PAT].max_n_segments = 1 << (30 - (22 - 5));
+      entry_size = sizeof(pat_node_large);
     } else {
       array_spec[SEGMENT_KEY].max_n_segments = GRN_PAT_MAX_N_SEGMENTS;
-      array_spec[SEGMENT_PAT].w_of_element = 4;
-      array_spec[SEGMENT_PAT].max_n_segments = 1 << (30 - (22 - 4));
+      entry_size = sizeof(pat_node);
+    }
+    segment_width_of_pat_entry = compute_bit_width_of_pat_entry(entry_size);
+    array_spec[SEGMENT_PAT].w_of_element = segment_width_of_pat_entry;
+    array_spec[SEGMENT_PAT].max_n_segments =
+      1 << (30 - (22 - segment_width_of_pat_entry));
+    if (flags & GRN_OBJ_KEY_WITH_SIS) {
+      entry_size = sizeof(sis_node) + value_size;
+    } else {
+      entry_size = value_size;
     }
-    array_spec[SEGMENT_SIS].w_of_element = w_of_element;
-    array_spec[SEGMENT_SIS].max_n_segments = 1 << (30 - (22 - w_of_element));
+    segment_width_of_pat_entry = compute_bit_width_of_pat_entry(entry_size);
```

Comment:
The `segment_width_of_pat_entry` is updated here. But the previous code wasn't updated. Is it intentional change?

Reply:
Yes. this is intentional change.

`array_spec[SEGMENT_PAT].w_of_element` and `array_spec[SEGMENT_SIS].w_of_element`are different values.
These values compute in `compute_bit_width_of_pat_entry()`, but `entry_size` of the argument of this function is different if we handle `SEGMENT_PAT` and `SEGMENT_SIS`

We need to update `segment_width_of_pat_entry` because of the process of compute`array_spec[SEGMENT_PAT].w_of_element` and array_spec[SEGMENT_SIS].w_of_element were Commonalized.